delivery_postlogistics/postlogistics/web_service.py

```diff
@@ -0,0 +1,516 @@
+            recipient["name2"] = partner.parent_id.name
+            recipient["personallyAddressed"] = False
+
+        # Phone and / or mobile should only be displayed if instruction to
+        # Notify delivery by telephone is set
+
+        picking_packagings = picking._get_picking_postlogistic_packages().mapped(
+            "packaging_id"
+        )
+        is_phone_required = [
+            packaging
+            for packaging in picking_packagings
+            if packaging.postlogistics_option_code == "ZAW3213"
+        ]
+
+        if is_phone_required:
+            phone = picking.delivery_phone or partner.phone
+            if phone:
+                recipient["phone"] = phone
+
+            mobile = picking.delivery_mobile or partner.mobile
+            if mobile:
+                recipient["mobile"] = mobile
+
+        return recipient
+
+    def _prepare_customer(self, picking):
+        """ Create a ns0:Customer as a dict from picking
+
+        This is the Postlogistic Customer, thus the sender
+
+        :param picking: picking browse record
+        :return a dict containing data for ns0:Customer
+
+        """
+        company = picking.company_id
+        partner = company.partner_id
+
+        customer = {
+            "name1": partner.name,
+            "street": partner.street,
+            "zip": partner.zip,
+            "city": partner.city,
+            "country": partner.country_id.code,
+            "domicilePostOffice": picking.carrier_id.postlogistics_office,
+        }
+        logo = picking.carrier_id.postlogistics_logo
+        if logo:
+            logo_image = Image.open(BytesIO(base64.b64decode(logo)))
+            logo_format = logo_image.format
+            customer["logo"] = logo.decode()
+            customer["logoFormat"] = logo_format
+        return customer
+
+    def _get_label_layout(self, picking):
+        """
+        Get Label layout define in carrier
+        """
+        return picking.carrier_id.postlogistics_label_layout.code
+
+    def _get_output_format(self, picking):
+        """
+        Get Output format define in carrier
+        """
+        return picking.carrier_id.postlogistics_output_format.code
+
+    def _get_image_resolution(self, picking):
+        """
+        Get Output Resolution Code define in carrier
+        """
+        return picking.carrier_id.postlogistics_resolution.code
+
+    def _get_license(self, picking):
+        """ Get the license
+
+        Take it from carrier and if not defined get the first license.
+
+        :return: license number
+        """
+        franking_license = picking.carrier_id.postlogistics_license_id
+        return franking_license.number
+
+    def _prepare_attributes(
+        self, picking, package=None, pack_num=None, pack_total=None
+    ):
+        services = []
+        if package:
+            postlogistics_packagings = package.mapped("packaging_id")
+            services = [
```

Comment:
services = package.packaging_id.split(",")

Reply:
updated, thanks for your feedback.